docs/reference/proving-authentication.mdx

```diff
@@ -5,190 +5,174 @@ title: Proving Ownership of a Flow Account
+### Authenticating a user using `account-proof`
+
+In order to authenticate your users via a wallet provider's account-proof service, your application needs to
+configure FCL by setting `fcl.accountProof.resolver` and providing two pieces of information.
+
+The `fcl.accountProof.resolver` is an async resolver function used by FCL to retrieve account proof data
+from your application server. It can be set in your application configuration under the `fcl.accountProof.resolver`
+key. The resolved data should include a specific application identifier (`appIdentifier`) and a random `nonce`.
+This data will be sent to the wallet for signing by the user. If the user approves and authentication is successfull,
+a signature is returned to the client in the data field of an `account-proof` service.
+
+**Application Identifier**
+
+An application identifier is a human-readable string that uniquely identifies your application name.
+The identifier is displayed by wallets when users are asked to approve a signing request.
+It helps users compare against the request origin and detect some malicious phishing attempts,
+improving trust of the application and signing process.
 
-The suggested order of operations of how your application might use
-`account-proof` begins with:
+**Random Nonce**
 
-- Optional: Your application may choose to configure FCL with an application
-  specific Domain Separation Tag. A custom configured Domain Separation Tag helps scope the cryptographic proof
-  generated for the `account-proof` service to your application. A Domain Separation Tag is a valid string less 
-  than 32 bytes long. (Note, inclusion of a Domain Separation Tag does not help to prevent Man-in-the-middle attacks.)
-  Your application can specify a custom Domain Separation Tag via `config.fcl.appDomainTag`.
+In addition to the `appIdentifier` your application must provide a **minimum 32-byte random nonce** as a hex string.
 
-```javascript
+If for any reason your application backend does not want to request an `account-proof` during authentication,
+it should send a response of `null`. If FCL receives a `null` response from the `accountProof.resolver` it will
+continue the authentication process with the wallet but will not request an account-proof and no signature will be returned.
+
+> In the case of a network or server error FCL will cancel the authentication process and return a rejected promise.
+
+```js
 import {config} from "@onflow/fcl"
 
+type AccountProofData {
+  // e.g. "Awesome App (v0.0)" - A human readable string to identify your application during signing
+  appIdentifier: string;  
+
+  // e.g. "75f8587e5bd5f9dcc9909d0dae1f0ac5814458b2ae129620502cb936fde7120a" - minimum 32-byte random nonce as hex string
+  nonce: string;          
+}
+
+type AccountProofDataResolver = () => Promise<AccountProofData | null>;
+
 config({
-  "fcl.appDomainTag": "AWESOME-APP-V0.0-user",
+  "fcl.accountProof.resolver": accountProofDataResolver
```

Comment:
What is `accountProofDataResolver` in this case?
I get that it is of type `AccountProofDataResolver` but I don't get from these docs how to code my own.

Can someone @justinbarry @gregsantos @chasefleming @10thfloor @JeffreyDoyle provide an example implementation here please?

Reply:
Hi @louisguitton 

_The fcl.accountProof.resolver is an async resolver function used by FCL to retrieve account proof data from your application server._ 

In this case `accountProofDataResolver` is this async function and can be implemented as you see fit to make a request to your app's backend. It will be called by FCL (if it is set) during the authentication process, and should return   the specified data described by the [account-proof docs](https://github.com/onflow/fcl-js/blob/master/docs/reference/proving-authentication.mdx#authenticating-a-user-using-account-proof).
 
_The resolved data should include a specific application identifier (appIdentifier) and a random nonce. This data will be sent to the wallet for signing by the user._

```js
const accountProofDataResolver = async () => ({
  appIdentifier: "Awesome App (v0.0)",
  nonce: "3037366134636339643564623330316636626239323161663465346131393662",
})
```

Reply:
thanks a lot for the answer @gregsantos . I ended up hardcoding like you suggested, but for some reason my user object doesn't return the service of type `account-proof`; only this

```json
{
    "f_type": "USER",
    "f_vsn": "1.0.0",
    "addr": "0xb5b04bfc7081d832",
    "cid": "ed88668b91e2995c2ebaa3676f6f676c652d6f61757468327c313039333437353734393039323532353031353731",
    "loggedIn": true,
    "expiresAt": 0,
    "services": [
        {
            "f_type": "Service",
            "f_vsn": "1.0.0",
            "type": "authn",
            "uid": "dapper#authn",
            "endpoint": "https://staging.accounts.meetdapper.com/fcl/authn-restricted",
            "id": "google-oauth2|109347574909252501571",
            "provider": {
                "address": "0x668b91e2995c2eba",
                "name": "Dapper Wallet",
                "icon": "https://meetdapper.com/logos/logo_dapper_new.png"
            }
        },
        {
            "f_type": "Service",
            "f_vsn": "1.0.0",
            "type": "pre-authz",
            "uid": "dapper#pre-authz",
            "endpoint": "https://staging.accounts.meetdapper.com/fcl/pre-authz-restricted",
            "method": "POP/RPC",
            "identity": {
                "f_type": "Identity",
                "f_vsn": "1.0.0",
                "address": "0xb5b04bfc7081d832",
                "keyId": 10
            },
            "params": {
                "key": "id",
                "value": "google-oauth2|999999999",
                "__typename": "Data"
            }
        },
        {
            "f_type": "Service",
            "f_vsn": "1.0.0",
            "type": "authz",
            "uid": "dapper#authz",
            "endpoint": "https://staging.accounts.meetdapper.com/api/fcl/authz",
            "method": "HTTP/POST",
            "identity": {
                "f_type": "Identity",
                "f_vsn": "1.0.0",
                "address": "0xb5b04bfc7081d832",
                "keyId": 10
            },
            "params": {
                "key": "id",
                "value": "google-oauth2|999999999",
                "__typename": "Data"
            }
        }
    ]
}
```

So for now I am stuck there but that might be an issue specific with Dapper Wallet, not with those docs. So I'll ask the Dapper Wallet team.
Thanks for help!